IoTHubBuddy/IoTViewer/AccountManager.cs

```diff
@@ -0,0 +1,362 @@
+﻿using Microsoft.Azure.Devices;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Data.Json;
+using Windows.Security.Authentication.Web.Core;
+using Windows.Security.Credentials;
+using Windows.Storage;
+using Windows.UI.ApplicationSettings;
+using Newtonsoft.Json;
+using System.Diagnostics;
+
+namespace IoTHubBuddy
+{
+    class AccountManager
+    {
+        const string AppSpecificProviderId = "myproviderid";
+        const string AppSpecificProviderName = "App specific provider";
+
+        const string AccountsContainer = "accountssettingscontainer";
+        const string ProviderIdSubContainer = "providers";
+        const string AuthoritySubContainer = "authorities";
+
+        const string MicrosoftProviderId = "https://login.microsoft.com";
+        const string MicrosoftAccountAuthority = "consumers";
+        const string AzureActiveDirectoryAuthority = "organizations";
+
+        const string MicrosoftAccountClientId = "none";
+        const string MicrosoftAccountScopeRequested = "wl.basic";
+        const string AzureActiveDirectoryClientId = "5cec90d8-14eb-41ac-9b2c-b3e8a7d95da5";
+        const string AzureActiveDirectoryScopeRequested = "wl.basic";
+        private static MainPage mp;
+        /// <summary>
+        /// build the account settings pane with the cached accounts
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        /// <param name="page"></param>
+        public static async void OnAccountCommandsRequested(
+           AccountsSettingsPane sender,
+           AccountsSettingsPaneCommandsRequestedEventArgs e, MainPage page)
+        {
+            mp = page;
+            // In order to make async calls within this callback, the deferral object is needed
+            AccountsSettingsPaneEventDeferral deferral = e.GetDeferral();
+            CreateLocalDataContainers();
+            await AddWebAccountProvidersToPane(e);
+            await AddWebAccountsToPane(e);
+
+            deferral.Complete();
```

Comment:
the deferral is there to delay showing the AccountsSettingsPane until I'm finished building it